indexer/indexer.go

```diff
@@ -45,7 +47,79 @@ func New(paths []string, coverpath *string, exts []string) (*Indexer, error) {
 		cp = &p
 	}
 
-	return &Indexer{paths: paths, coverpath: cp, exts: exts}, nil
+	return &Indexer{paths: paths, coverpath: cp, exts: exts, seen: NewSeenCache()}, nil
+}
+
+func (i *Indexer) Load() error {
```

Comment:
Some of this code may fit better as part of the cache. Ideally, the cache would handle it's own loading, and the indexer would query the cache during the indexing.